lambdas/LR_21_split_dps_extract/split_dps_extract.py

```
@@ -0,0 +1,211 @@
+
+        return split_dps_extract_handler(upload_key)
+
+    except Exception as err:
+        msg = f"Unhandled error when processing supplementary data file in LR-21"
+        error_response = log_dynamodb_error(
+            "99999999-2121-2121-2121-999999999999", "UNHANDLED_ERROR", msg
+        )
+
+        raise type(err)(error_response) from err
+
+
+def split_dps_extract_handler(upload_key: str) -> success:
+    """Handler function for DPS file split process
+
+    Args:
+        upload_key (str): A path to the s3 file key object
+
+    Returns:
+        success: A dict result containing a status and message
+    """
+
+    lines = read_file(upload_key)
+    LOG.info("Data lines extracted")
+
+    per_registered_gp = split_file(upload_key, lines)
+    LOG.info("Dictionary created for each registered GP")
+
+    write_files(per_registered_gp)
+    LOG.info("Output files for each registered GP was successful")
+
+    cleanup_files(upload_key, per_registered_gp)
+    LOG.info("Outdated GP file cleanup was successful")
+
+    return success(
+        f"LR-21 processed Supplementary data successfully, from file: {upload_key}"
+    )
+
+
+def read_file(upload_key: str) -> list:
+    """Retrieve and read supplementary file data
+
+    Args:
+        upload_key (str): A path to the s3 file key object
+
+    Returns:
+        list: A list of strings containing file data
+    """
+
+    try:
+        client = create_s3_client()
+
+        file_obj = client.get_object(Bucket=INPUT_BUCKET, Key=upload_key)
+
+        data_lines = file_obj["Body"].read().decode("utf-8").splitlines()
+
+        return data_lines
+
+    except (ClientError, UnicodeDecodeError) as err:
+        msg = f"LR-21 failed to read supplementary data from: {upload_key}"
+        error_response = log_dynamodb_error(
+            "99999999-2121-2121-2121-999999999999", "HANDLED_ERROR", msg
+        )
+
+        raise type(err)(error_response) from err

```

I forgot to say that we should actually not use type(err) because some exceptions require extra args - can you raise a custom exception or Exception instead? 

`Exception(error_response) from err`